POSIX: Implement pthread_condattr functions

Durchbruchswagen opened the pull request:
Added:
pthread_condattr_init
pthread_condattr_destroy
pthread_condattr_getclock
pthread_condattr_setclock

Fixes #59948
Fixes #59947

cfriedt commented on lib/posix/cond.c:
```diff
@@ -202,4 +202,39 @@ static int pthread_cond_pool_init(void)
+{
+	__ASSERT_NO_MSG(att != NULL);
+
+	att->clock = CLOCK_REALTIME;
```
I would recommend this is set to `CLOCK_MONOTONIC` by default.

Durchbruchswagen replied:
Done, changed default value to clock monotonic

cfriedt reviewed (commented):
Looks good. Just one code change and also please add tests in a separate second commit.

cfriedt commented on tests/posix/common/src/pthread.c:
```diff
@@ -233,6 +233,33 @@ void *thread_top_term(void *p1)
+	ret = pthread_condattr_init(&att);
+	zassert_equal(ret, 0, "pthread_condattr_init failed");
+
+	ret = pthread_condattr_getclock(&att, &clock_id);
+	zassert_equal(ret, 0, "pthread_condattr_getclock failed");
```
Would be good to apply this pattern elsewhere
```suggestion
	zassert_ok(pthread_condattr_getclock(&att, &clock_id));
```

Durchbruchswagen replied:
Done, replaced `zassert_equal` with `zassert_ok` wherever i could and removed `ret` variable

Durchbruchswagen committed:
POSIX: Implement pthread_condattr functions

Added:
pthread_condattr_init
pthread_condattr_destroy
pthread_condattr_getclock
pthread_condattr_setclock

Signed-off-by: Mateusz Marszalek <matti.marszalek@gmail.com>

Durchbruchswagen committed:
POSIX: Add tests for pthread_condattr

Add test coverage for init, destroy, setclock, getclock

Signed-off-by: Mateusz Marszalek <matti.marszalek@gmail.com>

cfriedt commented on tests/posix/common/src/pthread.c:
```diff
@@ -233,6 +233,28 @@ void *thread_top_term(void *p1)
 	return NULL;
 }
 
+ZTEST(posix_apis, test_pthread_condattr)
```
If you end up having to respin this for whatever reason, this should probably go into cond.c

cfriedt reviewed (approved):
LGTM - thanks for your contribution @Durchbruchswagen 👍 

Just in case there is another respin for whatever reason, could move the test to `cond.c`
